ui/src/map/layers/alignment/location-track-split-badge-layer.ts

```diff
@@ -65,63 +63,77 @@ const calculateSplitBounds = (
     ].filter((split) => split.name);
 };
 
+type LocationTrackSplitBadgeData = {
+    locationTracks: AlignmentDataHolder[];
+    startAndEnd: AlignmentStartAndEnd | undefined;
+};
+
+async function getLocationTrackSplitBadgeData(
+    publishType: PublishType,
+    splittingState: SplittingState | undefined,
+    changeTimes: ChangeTimes,
+    resolution: number,
+    mapTiles: MapTile[],
+): Promise<LocationTrackSplitBadgeData> {
+    if (resolution <= Limits.SHOW_LOCATION_TRACK_BADGES && splittingState) {
+        const [locationTracks, startAndEnd] = await Promise.all([
+            getSelectedLocationTrackMapAlignmentByTiles(
+                changeTimes,
+                mapTiles,
+                publishType,
+                splittingState.originLocationTrack.id,
+            ),
+            getLocationTrackStartAndEnd(
+                splittingState.originLocationTrack.id,
+                publishType,
+                changeTimes.layoutLocationTrack,
+            ),
+        ]);
+        return { locationTracks, startAndEnd };
+    } else {
+        return { locationTracks: [], startAndEnd: undefined };
+    }
+}
```

Comment:
Näistä toki saisi ehkäpä vielä vähän nätinpää (omaan silmään ainakin), jos kääntäisi iffittelyn ja käyttäisi early-returnia, jolloin yksi pykälä tuota sisennystä katoaisi varsinaisesta rungosta (sama toimisi siis useammassa paikassa kun näitä tässä kurkkailin):

```
async function getLocationTrackSplitBadgeData(
    ....
): Promise<LocationTrackSplitBadgeData> {
    if (resolution > Limits.SHOW_LOCATION_TRACK_BADGES || splittingState === undefined) {
        return { locationTracks: [], startAndEnd: undefined };
    }
    
    const [locationTracks, startAndEnd] = await Promise.all([
        getSelectedLocationTrackMapAlignmentByTiles(
            ...
        ),
        getLocationTrackStartAndEnd(
            ...
        ),
    ]);
    
    return { locationTracks, startAndEnd };
}
```

Reply:
Hmm, makuasiaa. Omaan silmään early return on melkein aina rumempi, mukaanlukien tässä. Tuohonkin versioon haluaisin ehdottomasti pistää iffin jälkeisen osion elseen, jolloin kunkin iffin haaran voi mieltää omana kokonaisuutenaan ja kukin niistä palauttaa. Jos sisennys tulee ongelmaksi, niin selvästi funkkari tekee liikaa asioita -> elsen (tai iffin) sisällön voi funktioida erikseen, mutta tässä tuo ei minusta ole vielä tarpeen.

Reply:
Omana mielipiteenä early-returnit auttavat tulkintaa, kun happy-path on lineaarisesti nähtävissä eikä koko funktion toimintaa tarvitse muistaa (asioita voi heittää mentaalistackiltä veks jokaisen early-returnin jälkeen). Tosin varmaankin tottumiskysymys miten iffittelyjä lukee. Mun aivot tulkitsee, että kokonainen if-(elseif)-(elseif)-else blokki pitää lukea, tulkita ja päätellä kerralla (raskasta).

Toinen itselle mielenpainunut syy on juurikin tuo sisäkkäisten iffittelyn välttämäminen, joka tekee taas tulkinnasta raskasta: 

```
    if (...){
        if (...){
            if(...){
                if(...){
                    ...
                } else {
                    return ...
                }
            } else {
                return ...
            }
        } else {
            return ...
        }
    } else {
        return ...
    }
```
Tällaiselta toki välttyy funktioimalla asioita. Tosin suhteellisen pienten, lähtökohtaisesti samaan asiaan liittyvien koodiblokkien tulkinta saattaa taas mennä hankalammaksi blokkeja liiallisesti funktioimalla (funktioiden nimeäminen on vaikeaa, pitää hyppiä koodissa eri kohtien välillä sekä pitää jälleen yksi funkkarinimi+operaatiokombinaatio työmuistissa=raskasta).

Mutta kuten sanoitkin, puolensa kummassakin tyylissä ja selvästi tottumiskysymyksiä.